### ISSUE TITLE
Research porting parallax programming protocol v1.1

### ISSUE THREAD

**Issue by jacobrosenthal** [parallaxinc/Parallax-IDE#11]
http://forums.parallax.com/attachment.php?attachmentid=56837

**Comment by jacobrosenthal** [comment]
Bare prototype at
https://github.com/jacobrosenthal/bs2-programmer/

**Comment by PropGit** [comment]
Jacob,

I took a brief look at what you've done so far.  It looks like you've made
fantastic progress!

**Comment by jacobrosenthal** [comment]
[bs2-programmer](https://github.com/jacobrosenthal/bs2-programmer) has been released as [0.1.0](https://github.com/jacobrosenthal/bs2-programmer/releases/tag/v0.1.0) and published to npm as [bs2-programmer](https://www.npmjs.com/package/bs2-programmer)

We demod in in the browser in an older branch of this app https://github.com/parallaxinc/ChromeIDE/tree/bs2-programmer

This sprint didn't require any graphical connection to programming so it simply run from the background page console with the command

```
window.bs2.upload('/dev/tty.usbserial-A502BMUQ', console.log.bind(console))
```
